src/libs/ajax/User.ts

```diff
@@ -0,0 +1,168 @@
+import * as fp from 'lodash/fp';
+import { cloneDeep, flow, unset } from 'lodash/fp';
+import { Config } from '../config';
+import axios from 'axios';
+import { getApiUrl, fetchOk, fetchAny } from '../ajax';
+import { CreateDuosUserResponse, DuosUserResponse, UpdateDuosUserResponse } from 'src/types/responseTypes';
+import { CreateDuosUserRequest, UpdateDuosUserRequestV1, UpdateDuosUserRequestV2 } from 'src/types/requestTypes';
+import { AcknowledgementMap, ApprovedDataset, Dataset, DuosUser, SimplifiedDuosUser } from 'src/types/model';
+
+export const User = {
+  getMe: async (): Promise<DuosUserResponse> => {
+    const url = `${await getApiUrl()}/api/user/me`;
+    const res = await axios.get(url, Config.authOpts());
+    return res.data;
+  },
+
+  getById: async (id: number): Promise<DuosUserResponse> => {
+    const url = `${await getApiUrl()}/api/user/${id}`;
+    const res = await axios.get(url, Config.authOpts());
+    return res.data;
+  },
+
+  list: async (roleName: 'Admin' | 'SigningOfficial'): Promise<DuosUserResponse[]> => {
+    const url = `${await getApiUrl()}/api/user/role/${roleName}`;
+    const res = await fetchOk(url, Config.authOpts());
+    return res.json();
+  },
+
+  create: async (request: CreateDuosUserRequest): Promise<CreateDuosUserResponse> => {
+    const url = `${await getApiUrl()}/api/dacuser`;
+    try {
+      const res = await fetchOk(
+        url,
+        fp.mergeAll([
+          Config.authOpts(),
+          Config.jsonBody(request),
+          { method: 'POST' },
+        ])
+      );
+      if (res.ok) {
+        return res.json();
+      }
+    } catch (err) {
+      return false;
+    }
+  },
+
+  updateSelf: async (payload: UpdateDuosUserRequestV1): Promise<UpdateDuosUserResponse> => {
+    const url = `${await getApiUrl()}/api/user`;
+    // We should not be updating the user's create date, associated institution, or library cards
```

Comment:
Note that this is currently broken in dev: https://broadworkbench.atlassian.net/browse/DCJ-362